### PR TITLE
Draft of documentation

### PR DESCRIPTION
**Pull request by VascoVerissimo**

### CODE REVIEW

**Comment by laurentheirendt** [comment]
@oHunewald, feel free to add your comments. Thanks 👍

Looks good at first sight, @VascoVerissimo ⚽️

File: docs/src/api/functions.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+```@contents
+```
+
+## Index
+
+```@index
+```
+
+## I/O
```
Comment (laurentheirendt) [inline docs/src/api/functions.md]: Maybe name this section `Input`.

File: docs/src/api/functions.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+```@autodocs
+Modules = [GigaSOM]
+Pages = ["Fcs_helper.jl"]
+```
+
+## Batch/Parallel SOM
```
Comment (laurentheirendt) [inline docs/src/api/functions.md]: Maybe name this `Core`

File: docs/src/index.md [hunk 3]
```diff
@@ -1,6 +1,81 @@
 ![GigaSOM.jl](https://prince.lcsb.uni.lu/GigaSOM.jl/img/logo-GigaSOM.jl.png?maxAge=0)
 
-GigaSOM.jl
+GigaSOM.jl - Huge-scale, high-performance flow cytometry clustering in Julia
 =========
 
-Huge-scale, high-performance flow cytometry clustering
+# Introduction
+
+Flow cytometry clustering for several hundred million cells has long been hampered by software implementations.
```
Comment (laurentheirendt) [inline docs/src/index.md]: May you format the text here properly?

File: docs/src/index.md [hunk 4]
```diff
@@ -1,6 +1,81 @@
 ![GigaSOM.jl](https://prince.lcsb.uni.lu/GigaSOM.jl/img/logo-GigaSOM.jl.png?maxAge=0)
```
Comment (laurentheirendt) [inline docs/src/index.md:1]: I think part of this could also go to the README file.

Comment (laurentheirendt) [reply docs/src/index.md:1]: And then part of it could be read in here (no duplication of text)

File: src/io/Fcs_helper.jl [hunk 5]
```diff
@@ -124,6 +124,10 @@ end
 """
     getMarkers(panel)
 
+Returns the lineage and functional markers on a given panel
+
+# Arguments:
+- `panel`: Panel table with a column for Lineage Markers and one for Functional Markers
```
Comment (laurentheirendt) [inline src/io/Fcs_helper.jl:112]: `lineage` and `functional`

File: src/parallel_som.jl [hunk 6]
```diff
@@ -285,6 +297,11 @@ end
     makeClassFreqs(som, vis, classes)
 
 Return a DataFrame with class frequencies for all neurons.
+
+# Arguments
+- `som`: a trained SOM
+- `vis`: index of the winner neuron for each training pattern in x
+- `classes`: Name of column with class information
```
Comment (laurentheirendt) [inline src/parallel_som.jl]: `name`

File: src/parallel_som.jl [hunk 7]
```diff
@@ -334,13 +351,23 @@ function makeClassFreqs(som, vis, classes)
 end
 
 
+
+# Arguments
+
+
 """
-    find_bmu(cod, sampl)
+    findBmu(cod, sampl)
```
Comment (laurentheirendt) [inline src/parallel_som.jl]: What are the input arguments here?

File: src/types.jl [hunk 8]
```diff
@@ -16,7 +16,7 @@ end
 
 
 """
-    struct Som
+    Som
```
Comment (laurentheirendt) [inline src/types.jl:19]: Why did you remove `struct`?

**Review by laurentheirendt — CHANGES_REQUESTED** [review]

File: docs/make.jl [hunk 9]
```diff
@@ -1,11 +1,27 @@
...
+        format = Documenter.HTML(prettyurls = !("local" in ARGS),
+                canonical = "https://lcsb-biocore.github.io/GigaSOM.jl/stable/",
+                assets = ["assets/icongigasom.ico"]),
+        sitename = "GigaSOM.jl",
+        authors = "The developers of GigaSOM.jl",
+        linkcheck = !("skiplinks" in ARGS),
+        pages = [
+                "Home" => "index.md",
+                "Tutorial" => "tutorials/tutorial.md",
+                "API" => Any[
```
Comment (laurentheirendt) [inline docs/make.jl]: Why `API`?

File: docs/make.jl [hunk 10]
```diff
@@ -1,11 +1,27 @@
-using Documenter
+using Documenter, GigaSOM
 
-makedocs(modules = [Documenter],
-         sitename="GigaSOM.jl")
+makedocs(modules = [GigaSOM],
+        clean = false,
+        format = Documenter.HTML(prettyurls = !("local" in ARGS),
+                canonical = "https://lcsb-biocore.github.io/GigaSOM.jl/stable/",
+                assets = ["assets/icongigasom.ico"]),
```
Comment (laurentheirendt) [inline docs/make.jl:7]: 🎉

File: docs/make.jl [hunk 11]
```diff
@@ -1,11 +1,27 @@
...
+                ],
+        )
 
 deploydocs(
     repo = "github.com/LCSB-BioCore/GigaSOM.jl.git",
     target = "build",
     branch = "gh-pages",
     devbranch = "origin/develop",
-)
+    deps = nothing,
```
Comment (laurentheirendt) [inline docs/make.jl]: Why `deps` and `make`?

File: docs/src/api/functions.md (hunk 1 above)
Comment (VascoVerissimo) [reply docs/src/api/functions.md]: done

File: docs/src/api/functions.md (hunk 2 above)
Comment (VascoVerissimo) [reply docs/src/api/functions.md]: done

File: src/parallel_som.jl [hunk 12]
```diff
@@ -167,7 +171,7 @@ function doEpoch(x::Array{Float64}, codes::Array{Float64}, dm::Array{Float64},
      # for each sample in dataset / trainingsset
      for s in 1:nRows
          sampl = vec(x[s, : ])
-         bmu_idx, bmu_vec = find_bmu(codes, sampl)
+         bmu_idx, bmu_vec = findBmu(codes, sampl)
```
Comment (laurentheirendt) [inline src/parallel_som.jl]: `idx, vec`

File: src/parallel_som.jl (hunk 7 above)
Comment (laurentheirendt) [reply src/parallel_som.jl]: I would rename this as `findBMU`

File: docs/make.jl (hunk 10 above)
Comment (VascoVerissimo) [reply docs/make.jl:7]: ;p

File: docs/make.jl (hunk 9 above)
Comment (laurentheirendt) [reply docs/make.jl]: `"Functions" => "functions.md",`

File: docs/src/index.md (hunk 4 above)
Comment (VascoVerissimo) [reply docs/src/index.md:1]: ok, we can do that separation once the text is complete so we can decide exactly what goes where

File: src/io/Fcs_helper.jl (hunk 5 above)
Comment (VascoVerissimo) [reply src/io/Fcs_helper.jl:112]: done

File: src/parallel_som.jl (hunk 6 above)
Comment (VascoVerissimo) [reply src/parallel_som.jl]: done

File: src/parallel_som.jl (hunk 7 above)
Comment (VascoVerissimo) [reply src/parallel_som.jl]: yeah I was talking about it with oliver this morning, there was a typo. taking care of it

File: docs/src/index.md (hunk 4 above)
Comment (laurentheirendt) [reply docs/src/index.md:1]: yup!

File: src/parallel_som.jl (hunk 12 above)
Comment (VascoVerissimo) [reply src/parallel_som.jl]: `bmuIdx`, `bmuVec` ?

Comment (VascoVerissimo) [reply src/parallel_som.jl]: so we dont lose track of what it is, because they are used in `doEpoch`

**Review by oHunewald — CHANGES_REQUESTED** [review]
I would cancel the Immunology part, someone looking for GigaSOM has already a good understanding in this topic. The explanation of som: you should mention that by using GigaSOM, which implements the batch som algorithm, we are not using a learning rate.

**Review by laurentheirendt — APPROVED** [review]

**Comment by laurentheirendt** [comment]
Looks good 👍 Well done.
